### PR TITLE
Commiting the advances for the Mortar in the last weeks

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by loumalouomega** [comment]
Sorry, I have been working so long in this branch, I promise that the next pull request will be smaller

I do the pull request concerning to #193 , where my mesh tying condition can be of interest

File: applications/MeshingApplication/custom_utilities/mmg_utility.h [hunk 1]
```diff
@@ -213,11 +213,11 @@ class MmgUtility
...
-           std::cout << "//---------------------------------------------------//" << std::endl;
-           std::cout << "//---------------------------------------------------//" << std::endl;
-           std::cout << "//---------------  BEFORE REMESHING   ---------------//" << std::endl;
-           std::cout << "//---------------------------------------------------//" << std::endl;
-           std::cout << "//---------------------------------------------------//" << std::endl << std::endl;
+            std::cout << "//---------------------------------------------------//" << std::endl;
+            std::cout << "//---------------------------------------------------//" << std::endl;
+            std::cout << "//---------------  BEFORE REMESHING   ---------------//" << std::endl;
+            std::cout << "//---------------------------------------------------//" << std::endl;
+            std::cout << "//---------------------------------------------------//" << std::cout << std::endl;
```
Comment (jcotela) [inline applications/MeshingApplication/custom_utilities/mmg_utility.h]: This looks strange...

Comment (loumalouomega) [reply applications/MeshingApplication/custom_utilities/mmg_utility.h]: It is the diff, not my fault

**Comment by jginternational** [comment]
Please, keep 
```"mesh_id"                   : 0```
In the alm_contact_process, for compatibility with the rest of processes

File: applications/MeshingApplication/custom_utilities/mmg_utility.h (hunk 1 above)
Comment (jcotela) [reply applications/MeshingApplication/custom_utilities/mmg_utility.h]: I'm referring to line 220: `std::cout << "//--...--//" << std::cout << std::endl;`
Why do you need two `std::cout` there?

Comment (loumalouomega) [reply applications/MeshingApplication/custom_utilities/mmg_utility.h]: Ohh, I see, thanks

**Comment by pooyan-dadvand** [comment]
Just as a comment: It would be great if you could divide this PR to some smaller ones by cherry picking your commits in some smaller new branches an make PR for each smaller branch

**Comment by loumalouomega** [comment]
@pooyan-dadvand I can do smaller branches in the future, but I prefer not to experiment with cherry picking with all my work of the last weeks (I have bad experiences with git in the past). Besides this is mainly my application, so it is not interfering with anything from the core or any other application

**Comment by pooyan-dadvand** [comment]
So at least I would suggest to stop committing more in this branch.

**Comment by loumalouomega** [comment]
Ok, done

**Pull request closed without merging**